### PR TITLE
Core data

### PR DESCRIPTION
**Pull request by lsdlabs**
## What was done:
I added save functionality to the ToDo app.

## How it was completed:
I created a save function that accepts a String which will hold a ToDo item.  

Re: the save function-
The persistentContainer property in the AppDelegate that is part of the Core Data stack is optional.  
To access it, I created a reference to the app delegate (which was guarded against since it is optional).  This was done to get access to an NSManagedObjectContext in order to save and retrieve information from the Core Data store.  (Note: This PR is only handling saving.)

Next, I inserted the managed object into the managed object context using the static method ‘entity(forEntityName:in:)’ which belongs to the NSEntityDescription class.  I then set the ‘item’ attribute using key-value coding with the key (‘item’) spelled exactly  as it appears in the data model (in the ToDo.xcdatamodeld file).

I called save on the managedContext object created inside of a do-try-catch because it can throw an error.  Afterwards, I appended the toDoListItem to the toDoList array so it can be displayed when the tableview reloads.

In the extension of ViewController, I modified the tableview cellForRowAt dataSource method to match cells with the corresponding NSManagedObject.

Last but not least, I called save from inside the saveAction (UIAlertAction).

## Concerns (Optional):
Not really a concern, but I was confused for a while about why after restarting the app I could not see the items I added to the list.  I learned that while items are being saved to disk, I have to ‘fetch’ them otherwise it will just sit there.  That feature will be available in the next PR.

## How you can test it:
Run the app and check if you can add items to the list.  If you choose to do a fresh app launch, you will notice the ToDoList array/tableview is empty

### CODE REVIEW

File: ToDo/ToDo/ViewController.swift [hunk 1]
```diff
@@ -27,13 +28,15 @@ class ViewController: UIViewController {
...
-                        self.tableView.reloadData()
+        let saveAction = UIAlertAction(title: "Save", style: .default) {
+            [unowned self] action in
+            
+            guard let textField = alert.textFields?.first,
+                let itemToSave = textField.text else {
+                    return
+            }
+            self.save(item: itemToSave)
+            self.tableView.reloadData()
```
Comment (northofnormal) [inline ToDo/ToDo/ViewController.swift:38]: 🍰Totally optional style thing, but: I would put the two `let`s on a single line, and put a new line between the closing `}` and `self.save...` I think it's more readable, but no big deal.

File: ToDo/ToDo/ViewController.swift [hunk 2]
```diff
@@ -46,12 +49,38 @@ class ViewController: UIViewController {
...
+        
+        let managedContext =
+            appDelegate.persistentContainer.viewContext
+        
+        let entity =
+            NSEntityDescription.entity(forEntityName: "ToDoList",
+                                       in: managedContext)!
+        
+        let toDoListItem = NSManagedObject(entity: entity,
+                                           insertInto: managedContext)
```
Comment (northofnormal) [inline ToDo/ToDo/ViewController.swift]: 🍰Same here, I'd put all these on one line. (Jeff Kelly, for example, would disagree with this.) It's 2019! We have wide-screen monitors! Spread out! We have space!

File: ToDo/ToDo/ViewController.swift [hunk 3]
```diff
@@ -60,11 +89,12 @@ extension ViewController: UITableViewDataSource {
     func tableView(_ tableView: UITableView,
                    cellForRowAt indexPath: IndexPath)
         -> UITableViewCell {
```
Comment (northofnormal) [inline ToDo/ToDo/ViewController.swift]: 🍰I AM CAKING THIS UNDER DURESS. But looking at it makes me itch. You do you, none of this is wrong. It's just _gross_ 🤣

**Review by northofnormal — APPROVED** [review]

File: ToDo/ToDo/ViewController.swift (hunk 3 above)
Comment (lsdlabs) [reply ToDo/ToDo/ViewController.swift]: 🍰 style changes made! 🎉

**Pull request merged**